### ISSUE TITLE
关于不更新BN的测试

### ISSUE THREAD

**Issue by justopit** [automan000/Camera-based-Person-ReID#10]
CBN是一个非常棒的idea，他在更新少量的参数情况下就能取得非常卓越的性能提升。为了复现你们论文的baseline，我在你们的test_mdel.py的49行之后，为每个camera更新BN之前增加了一个正常测试的函数，然而我发现这样子跑出来的性能会低于你们的baseline 10几个点以上。我想问一下，在copy模型的state dict的时候或者之后，是不是没有copy到某个东西？还是保存的模型的state dict有问题？
代码如下
`    

    def _normal_testing( data ):
            data_loader = DataLoader(
                data_manager.init_datafolder(opt.testset_name, data, TestTransform(opt.height, opt.width)),
                batch_size=opt.test_batch, num_workers=opt.workers,
                pin_memory=pin_memory, shuffle=False
            )
            fs, pids, camids = reid_evaluator.produce_features(data_loader, normalize=True)
            return fs, pids, camids

    print('Processing query features...')
    qf, q_pids, q_camids = _normal_testing(dataset.query )
    print( qf.shape )
    print('Processing gallery features...')
    gf, g_pids, g_camids = _normal_testing(dataset.gallery )
    print( gf.shape )
    print('Computing CMC and mAP...')
    reid_evaluator.get_final_results_with_features(qf, q_pids, q_camids, gf, g_pids, g_camids)`

**Comment by automan000** [comment]
感谢关注。这样操作是不能复现Baseline性能的。
Baseline的特点是：(1) 使用dataset统计量训练; (2) 使用该统计量测试。
您在这里使用了我们的训练代码，也就是每一个Camera分别做标准化。在这种条件下，BN记录的统计量是这些Camera统计量的加权平均。这种统计量与Baseline的计算结果不同，它是没有物理含义的，不应该被用于测试。
如果要复现Baseline，您还需要修改训练阶段的标准化方式。
